app/controllers/wpcc/engine_controller.rb

```diff
@@ -1,7 +1,22 @@
 module Wpcc
   class EngineController < Wpcc.parent_controller.constantize
     protect_from_forgery
+    before_action :check_session
 
     layout 'wpcc/engine'
+
+    def check_session
+      session_expired? ? reset_session : update_session_expiry
+    end
+
+    private
+
+    def session_expired?
+      session[:expires_at] && session[:expires_at] <= Time.current
+    end
+
+    def update_session_expiry
+      session[:expires_at] = Time.current + 30.minutes
```

Comment:
We had a discussion with Ed and he wanted a very short time period e.g. 15 minutes. I agree @norm's comment re people multi-tasking. The form is only 2 steps and requires minimal input which is why he agreed that it was sensible to up it to 30 minutes.

It is related to [this user story](https://moneyadviceservice.tpondemand.com/RestUI/Board.aspx?acid=74E24C42BB81286E55C11FA8BF7FAAF3#page=board/5624876525867731357&appConfig=eyJhY2lkIjoiREI4ODcwRjkxMDNDRTM2NTlBMzhDNTRBRTVBNUU1N0UifQ==&boardPopup=userstory/8408/silent) and I will update the description accordingly. 

I was looking for comments re approach to confirm if what I've done is ok or complete rubbish.